### PR TITLE
Build docker image including a native executable (DEV-99)

### PR DESCRIPTION
**Pull request by head1328**

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -11,19 +11,79 @@ on:
...
+        container-name: message-service
+        image: ${{ steps.build-image.outputs.FULL_IMAGE_NAME }}:${{ env.VERSION }}
+
+    - name: Deploy Amazon ECS task definition
+      uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+      with:
+        task-definition: ${{ steps.task-def.outputs.task-definition }}
+        service: message-service
+        cluster: dealog-dev
+        wait-for-service-stability: true
```
Comment (scflode) [inline .github/workflows/maven.yml]: This will go in a separate action, correct?

Comment (head1328) [reply .github/workflows/maven.yml]: yes it will.

Comment (head1328) [reply .github/workflows/maven.yml]: now it is

File: .github/workflows/build-and-publish.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
```
Comment (scflode) [inline .github/workflows/build-and-publish.yml:6]: We don't have a `main` branch (yet). But I guess this doesn't matter.

File: .github/workflows/build-and-publish.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      - develop
+  pull_request:
+    branches:
+      - develop
```
Comment (scflode) [inline .github/workflows/build-and-publish.yml:10]: Shouldn't the `main` branch then also go here?

File: .github/workflows/deploy-to-amazon-ecs.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+name: Deploy
+
+on:
+  workflow_run:
+    workflows: [ "Build" ]
+    branches:
+      - main
```
Comment (scflode) [inline .github/workflows/deploy-to-amazon-ecs.yml:7]: Same thing with the branch names here.

**Review by scflode — APPROVED** [review]
LGTM!

**Pull request merged**